homeassistant/components/isy994/climate.py

```diff
@@ -0,0 +1,360 @@
+
+        current_operation = next((i for i in self._node.aux_properties
+                                  if i['id'] == ISY_OPERATION_MODE), False)
+        if current_operation:
+            self._current_isy_operation = VALUE_TO_ISY_MODE.get(
+                current_operation['value'], None)
+            self._current_operation, self._current_hold_mode = \
+                VALUE_TO_HASS_MODE.get(current_operation['value'], None)
+
+        await super().async_added_to_hass()
+        self.async_schedule_update_ha_state()
+        self._node.controlEvents.subscribe(self._node_control_handler)
+
+    def _node_control_handler(self, event: object) -> None:
+        """Handle control events coming from the primary node.
+
+        The ISY does not report some properties on the root of the node,
+            they only show up in the event log:
+
+        ISY_FAN_MODE, ISY_OPERATION_STATE, ISY_UOM will be set the first
+            time the event is fired by the ISY for those controls.
+
+        Current Temperature is updated by PyISY in node.status and we don't
+            need to listen for it here.
+        """
+        if event.event == ISY_FAN_MODE:
+            self._fan_mode = FAN_VALUE_TO_MODE.get(int(event.nval),
+                                                   None)
+        elif event.event == ISY_OPERATION_STATE:
+            self._current_status = VALUE_TO_STATE.get(int(event.nval),
+                                                      None)
+        elif event.event == ISY_OPERATION_MODE:
+            self._current_isy_operation = \
+                VALUE_TO_ISY_MODE.get(int(event.nval), None)
+            self._current_operation, self._current_hold_mode = \
+                VALUE_TO_HASS_MODE.get(int(event.nval), None)
+        elif event.event == ISY_UOM:
+            if int(event.nval) == 1:
+                self._temp_unit = TEMP_CELSIUS
+            elif int(event.nval) == 2:
+                self._temp_unit = TEMP_FAHRENHEIT
+        elif event.event == ISY_CURRENT_HUMIDITY:
+            self._current_humidity = int(event.nval)
+        elif event.event == ISY_TARGET_TEMP_HIGH:
+            self._target_temp_high = self._fix_temp(int(event.nval))
+        elif event.event == ISY_TARGET_TEMP_LOW:
+            self._target_temp_low = self._fix_temp(int(event.nval))
+        self.schedule_update_ha_state()
+
+    @property
+    def state(self):
+        """Return the current state."""
+        return self._current_status
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_TENTHS
+
+    @property
+    def value(self):
+        """Get the current value of the device."""
+        # pylint: disable=protected-access
+        return self._fix_temp(self._node.status._val)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._temp_unit:
```

Comment:
How can we report current and target temperatures without knowing the unit of those values?

Reply:
I believe this is a limitation of the ISY and how it handles climate devices--not all devices report back a unit and those that do don't report it on initial query, it only comes through on the event log after connection.  In the code I assume the user is using the same units in HA as the ISY, until the ISY actually passes the UOM.